main_test.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	firecracker "github.com/firecracker-microvm/firecracker-go-sdk"
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+)
+
+func TestParseBlockDevices(t *testing.T) {
+	tempFile, err := ioutil.TempFile("", "firectl-test-drive-path")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Remove(tempFile.Name())
+	defer tempFile.Close()
+	validDrive := models.Drive{
+		DriveID:      firecracker.String("2"),
+		PathOnHost:   firecracker.String(tempFile.Name()),
+		IsReadOnly:   firecracker.Bool(false),
+		IsRootDevice: firecracker.Bool(false),
+	}
+	cases := []struct {
+		in          []string
+		outDrives   []models.Drive
+		expectedErr func(error) bool
+	}{
+		// no suffix
+		{
+			in:        []string{"/path"},
+			outDrives: nil,
+			expectedErr: func(a error) bool {
+				return a == invalidDriveSpecificationNoSuffix
+			},
+		},
+		// no path
+		{
+			in:        []string{":rw"},
+			outDrives: nil,
+			expectedErr: func(a error) bool {
+				return a == invalidDriveSpecificationNoPath
+			},
+		},
+		// path does not exist
+		{
+			in:          []string{"/does/not/exist:ro"},
+			outDrives:   nil,
+			expectedErr: os.IsNotExist,
+		},
+		// valid
+		{
+			in:        []string{tempFile.Name() + ":rw"},
+			outDrives: []models.Drive{validDrive},
+			expectedErr: func(a error) bool {
+				return a == nil
+			},
+		},
+	}
+	for _, c := range cases {
+		drives, err := parseBlockDevices(c.in)
+		if !equalDrives(c.outDrives, drives) {
```

Comment:
would probably be better to just use `reflect.DeepEqual` here.

Reply:
Yea that is definitely a lot better, thanks!